swap_meet/vendor.py

```diff
@@ -1,2 +1,77 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory= None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item  
+        return None
+
+    def get_by_category(self, category):
+        # list to store sorted data
+        things_in_category = []
+        # loop iterates through inventory to find matching category
+        for thing in self.inventory:
+            if thing.category == category:
+                things_in_category.append(thing)
+        return things_in_category
+    
+    def swap_items(self, vendor_2, item_1, item_2):
+        if item_1 in self.inventory and item_2 in vendor_2.inventory:
+        # remove first item from first vendor and give to other
+            self.inventory.remove(item_1)
+            vendor_2.inventory.append(item_1)
+        # remove second item from second vendor and give to other
+            vendor_2.inventory.remove(item_2)
+            self.inventory.append(item_2)
+            return True
+        return False
+
+    def swap_first_item(self, other_vendor):
+        # ensures there are any items to swap
+        if self.inventory == [] or other_vendor.inventory == []:
```

Comment:
A more pythonic way to write this would be to check for the falseyness: `if not self.inventory or not other_vendor.inventory:`